state_machine/drivers/pycubedmini/lib/pycubed.py

```diff
@@ -262,394 +306,168 @@ def reinit(self, dev):
-        save the passed dataset to the passed savefile
-        dataset should be a set of lists; each line is a list:
-            save(([line1],[line2]))
-        to save a string, make it an item in a list:
-            save(['This is my string'])
-        by default, savefile is not passed
-        """
-        # if no savefile is passed, use the current filename attribute
-        # by default
-        if savefile is None:
-            savefile = self.filename
-
-        # open save file
+
+# initialize Satellite as cubesat
+cubesat = Satellite()
+
+"""
+IMU-related functions
+TODO: cubesat.imu hardware check
+"""
+
+def acceleration():
+    """ return the accelerometer reading from the IMU """
+    return cubesat.IMU.accel
+
+def magnetic():
+    """ return the magnetometer reading from the IMU """
+    return cubesat.IMU.mag
+
+def gyro():
+    """ return the gyroscope reading from the IMU """
+    return cubesat.IMU.gyro
+
+def temperature():
+    """ return the thermometer reading from the IMU """
+    return cubesat.IMU.temperature  # Celsius
+
+
+"""
+Burnwire-related functions
+TODO: cubesat.burnwire1 hardware check
+"""
+
+def burn(burn_num='1', dutycycle=0, freq=1000, duration=1):
+    """
+    control the burnwire(s)
+    initialize with burn_num = '1' ; burnwire 2 IC is not set up
+    """
+    # BURN1 = -Z,BURN2 = extra burnwire pin, dutycycle ~0.13%
+    dtycycl = int((dutycycle / 100) * (0xFFFF))
+
+    # print configuration information
+    print('----- BURN WIRE CONFIGURATION -----')
+    print(f'\tFrequency of: {freq}Hz')
+    print(f'\tDuty cycle of: {100 * dtycycl / 0xFFFF}% (int:{dtycycl})')
+    print(f'\tDuration of {duration}sec')
+
+    # initialize burnwire based on the burn_num passed to the function
+    if '1' in burn_num:
+        burnwire = cubesat.burnwire1
+    elif '2' in burn_num:
+        return False  # return False because burnwire 2 IC is not set up
+        # burnwire = self.burnwire2
+    else:
+        return False
+
+    set_RGB(255, 0, 0)  # set RGB to red
+
+    # set the burnwire's dutycycle; begins the burn
+    burnwire.duty_cycle = dtycycl
+    time.sleep(duration)  # wait for given duration
+
+    # set burnwire's dutycycle back to 0; ends the burn
+    burnwire.duty_cycle = 0
+    set_RGB(0, 0, 0)  # set RGB to no color
+
+    cubesat._deployA = True  # sets deployment variable to true
+    burnwire.deinit()  # deinitialize burnwire
+
+    return cubesat._deployA  # return true
+
+
+"""
+Radio related functions
+TODO: cubesat.radio hardware check
+Keeping the aliasing so we can do a hardware check for cubesat.radio
+"""
+
+def send(data, *, keep_listening=False, destination=None, node=None,
+         identifier=None, flags=None):
+    """ Wrap cubesat.radio.send to allow for hardware checks """
+    cubesat.radio.send(data, keep_listening=keep_listening,
+                       destination=destination, node=node, identifier=identifier,
+                       flags=flags)
+
+def listen():
```

Comment:
listen=cubesat.radio.listen

Reply:
Forgot python had this quirk; that would save a lot of lines!